dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/ShapefileHandler.java

```diff
@@ -365,7 +372,20 @@ private void examineZipFile() {
                 if (fileNamesInZip.contains(unzipFileName)) {
                     throw new IllegalStateException("Found file-name collision: " + unzipFileName);
                 }
+
+                if (fileSizeLimit != null && zipFileEntry.getSize() >= fileSizeLimit) {
+                    throw new FileExceedsMaxSizeException(fileSizeLimit,
+                            MessageFormat.format(BundleUtil.getStringFromBundle("file.addreplace.error.file_exceeds_limit"),
+                            bytesToHumanReadable(fileSizeLimit)));
```

Comment:
Since we just catch and log exception in `FileTypeDetector` I don't know if it is worth to have some "pretty" exception messages. It just only causes confusion. `BundleUtil.getStringFromBundle` depends on currently logged user. Once the message will be in English and once in Polish. Maybe some standard exception will be enough?

Reply:
Fair point, changed it to `throw new IngestException(IngestError.UNZIP_SIZE_FAIL)` which is used also by processing of normal zip files.